bindings/matlab/libiio_if.m

```diff
@@ -398,7 +398,10 @@ function delete(obj)
             % Load the libiio library
             if(~libisloaded(obj.libname))
                 try
+                    % ignore unknown type warnings due to header parsing limitations
+                    warning('off', 'MATLAB:loadlibrary:TypeNotFound');
                     [notfound, warnings] = loadlibrary(obj.libname, obj.hname, 'addheader', 'iio.h');
+                    warning('on', 'MATLAB:loadlibrary:TypeNotFound');
                 catch exception
```

Comment:
Right, the best practice here would be:
```matlab
try
    % ignore unknown type warnings due to header parsing limitations
    warnState = warning('off', 'MATLAB:loadlibrary:TypeNotFound');
    cleanupObj = onCleanup(@()warning(warnState));
    [notfound, warnings] = loadlibrary(obj.libname, obj.hname, 'addheader', 'iio.h');
    cleanupObj = []; % restore the warning state
```

The onCleanup() function returns an object that will execute the specified function when it is cleared. We explicitly clear it after loading the library, but it will be implicitly cleared when you return from the function in the catch block.